### PR TITLE
Feat/1386 browser hello world test

### PR DESCRIPTION
**Pull request by GlenDC**
Add browser hello testplan and integration tests, making use of playwright and docker:generic,
in function of issue #1386 

Feature/Requirement checklist:

- [X] hello browser testplan 
- [x] hello browser integration test
    - [x] success integration test
    - [x] failure integration test

Usage example of hello world testplan:

```
testground plan import --from ./plans/example-browser
```

```
testground run single \
    --plan example-browser --testcase success \
    --instances 1 \
    --builder docker:generic --runner local:docker \
    --tp browser=firefox --wait
```

You can run firefox/chromium for both success and failure cases using the single integration test:

```
./integration_tests/example_04_browser.sh
```

~~Secondary Requirement checklist:~~

> Question: should we do that as part of this PR, or a new one?
>> Answer: a new PR

- ~~use testground sdk directly from browser~~
- ~~add test plan to showcase how to do this~~
- ~~add integration tests which makes use of this testplan~~

### CODE REVIEW

**Comment by GlenDC** [comment]
Placebo integration test seems to fail, which is unrelated from the work done for this PR?

File: plans/example-browser/playwright-runner.js
Comment (brdji) [inline plans/example-browser/playwright-runner.js:18]: Is this the standard way of checking which browser to open? 
I would prefer a simple string check, and if the user inputs something like "opera", we should return an error.

File: plans/example-browser/playwright-runner.js
Comment (brdji) [inline plans/example-browser/playwright-runner.js:3]: Since this function forms the base of the tests, I would like to see comments on its behavior and usage; same goes for `runTestFn`

File: plans/example-browser/playwright-runner.js
Comment (GlenDC) [reply plans/example-browser/playwright-runner.js:18]: No, not standard. Just my way of integrating that configurability into testground using a simple regex (to not have to worry about surrounding whitespaces or case sensitivity). So there are two things here:

1. require an exact string (be it case insensitive or not)
2. also expect for chromium an exact string, which would be the default

I can apply both things. Which I'll assume for now is what you want.

File: plans/example-browser/failure.js
Comment (brdji) [inline plans/example-browser/failure.js:11]: I would tweak the Error / success messages to signify that the expected behavior is for this test to fail, and not reach the end of the function.

File: plans/example-browser/playwright-runner.js
Comment (GlenDC) [reply plans/example-browser/playwright-runner.js:3]: Sure, is there a particular format of docstring you expect here? Wasn't under the impression that this needed any comments given the fact that it's (a) an example and (b) very small in scope. But sure, it also cannot hurt, especially for an example. So I'll add it with pleasure.

runFn is a private function though, but I'll never the less add a comment.

Note that for the second example (the one I still need to add and which will run testground-sdk from within the browser) the flow will still be a bit different and slightly more complex)

File: plans/example-browser/playwright-runner.js
Comment (brdji) [reply plans/example-browser/playwright-runner.js:18]: Ignoring whitespaces or case sensitivity sounds like a good idea; I would apply both points.

File: plans/example-browser/playwright-runner.js
Comment (brdji) [reply plans/example-browser/playwright-runner.js:3]: The functions are clear enough, but I have future developers in mind - when someone wishes to use testground browser tests to their own project, they will use this example (or others, if we replace this). It makes sense to keep the example as clear and documented and possible, so in the future, people don't have to question why / how something was done.

Does this make sense?

File: plans/example-browser/failure.js
Comment (brdji) [reply plans/example-browser/failure.js:11]: If the `failure.js` is integrated into the CI, or if someone runs the test on their own, the messages can be a bit confusing. Here's what I would do: the default output should be something like 
`failure.js: If you see this message, that means the test failed as expected. User agent is ${}.`
And for the failure
`failure.js: If you see this message, something bad has occured - this test is meant to fail`

That way, if I see the message `failure.js - PASSED X` (or similar), it won't be as confusing.
Thoughts?

Comment (laurentsenta) [reply plans/example-browser/failure.js:11]: Let's simplify and use the same code that we used in `failure.js`
it's explicit and self-documented.
https://github.com/testground/testground/blob/62c18ac2150aae5f49a091b03150872440e8c564/plans/example-js/failure.js

File: plans/example-browser/playwright-runner.js
Comment (laurentsenta) [reply plans/example-browser/playwright-runner.js:18]: Both are for me in this version,

In future iterations, let's favor exact string matches when we can: It's hard to plan for how a parameter might evolve and what edge cases might bite us in the future. (remember windows 9? :)

File: plans/example-browser/playwright-runner.js
Comment (laurentsenta) [reply plans/example-browser/playwright-runner.js:3]: Thanks for the note @brdji, huge 👍 to aiming for more documentation and more comments,

A few notes to make sure we're staying on the pragmatic side of things:

- We expect this code to change a lot over the next few weeks, so it's fine to hold off "user-oriented" documentation (this is the `testground:edge` version). A `README.md` would be a great start with an example or two coming from the integration test, no blockers for me.
- We need "maintainers-oriented" comments when a piece of code hides some specialized knowledge or some context that cannot be guessed by a maintainer having a quick read. If we feel like a piece of code is self-explanatory, let's not worry about comments.

**Comment by laurentsenta** [comment]
The CI issues should have been fixed, could you rebase on `master` @GlenDC ?

**Review by laurentsenta — APPROVED** [review]

File: plans/example-browser/playwright-runner.js
Comment (GlenDC) [reply plans/example-browser/playwright-runner.js:18]: Resolved as an exact string, as requested.

File: plans/example-browser/playwright-runner.js
Comment (GlenDC) [reply plans/example-browser/playwright-runner.js:3]: Added the desired comments.

File: plans/example-browser/failure.js
Comment (GlenDC) [reply plans/example-browser/failure.js]: Went with @brdji's proposal, seems the most fair IMHO.

**Comment by GlenDC** [comment]
> @GlenDC thanks for the ping, could you rebase over the latest master? I'll make sure this goes in next.
> @brdji could you approve the PR if you are satisfied with the latest changes?

Ok. Master rebased. Ready for approval and merge.
Locally I already am working on step 2 and I'll open a Draft PR once this one is merged in.

**Review by brdji — APPROVED** [review]
Good job!

**Comment by GlenDC** [comment]
@laurentsenta ball is in your camp again, as I do not have merge rights. All good now AFAIK.

**Comment by mxinden** [comment]
Thanks for all the work here.

//CC @achingbrain for visibility.

**Pull request merged**
